tests/setup.py

```diff
@@ -151,6 +151,17 @@ def ensure_document_listener_created(self) -> bool:
         return False
 
     def await_message(self, method: str, promise: Optional[YieldPromise] = None) -> 'Generator':
+        """
+        Awaits until the server responds with data returned for a request with specified method.
+        If server has already received a request with specified method before, it will immediately
+        return the response for that previous request. If it hasn't received such request yet, it
+        will wait for it and then respond or we're time out.
```

Comment:
This is really tricky to describe properly. I did my best and it's probably still a mess.

Reply:
It's not entirely correct either. It awaits the moment when the server has received any kind of payload with the given method. Not when it responds.

Reply:
I think by saying "until the server responds" I meant when it responds to this test request, not the original request.

Reply:
Improved it a bit.